internal/Get-DbaADObject.ps1

```diff
@@ -24,108 +29,65 @@ This program is distributed in the hope that it will be useful, but WITHOUT ANY
 You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 .EXAMPLE
-Get-DbaADObject -ADObject "contoso\(&(objectCategory=User)(sAMAccountName=ctrlb))"
+Get-DbaADObject -ADObject "contoso\ctrlb" -Type User
+
+Seaches in the contoso domain for a ctrlb user
+
+.EXAMPLE
+Get-DbaADObject -ADObject "contoso\sqlcollaborative" -Type Group
 
-Seaches in the contoso domain for a ctrlb sAMAccountName
+Seaches in the contoso domain for a sqlcollaborative group
+
+.EXAMPLE
+Get-DbaADObject -ADObject "contoso\sqlserver2014$" -Type Group
+
+Seaches in the contoso domain for a sqlserver2014 computer (remember the ending $ for computer objects)
 
 #>
 	[CmdletBinding()]
 	Param (
-		[string[]]$ADObject
+		[string[]]$ADObject,
+		[ValidateSet("User","Group","Computer")]
+		[string]$Type,
+		[switch]$Silent
 	)
-
 	BEGIN {
-		function ConvertTo-Dn ([string]$dns)
-		{
-			$array = $dns.Split(".")
-			for ($x = 0; $x -lt $array.Length; $x++)
-			{
-				if ($x -eq ($array.Length - 1)) { $separator = "" }
-				else { $separator = "," }
-				[string]$dn += "DC=" + $array[$x] + $separator
-			}
-			return $dn
+		try {
+			Add-Type -AssemblyName System.DirectoryServices.AccountManagement
+		} catch {
+			write-warning 'Failed to load the required module'
```

Comment:
I've come across this on other commands I've looked at, certain places original author used Write-Warning and would need to be replaced with `Stop-Function` and not just `Write-Message -Warning`. We will have to look at those more closely to make sure each is used properly.